src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RemoveEmptyFinalizersTests.cs

```diff
@@ -216,6 +218,86 @@ public class Class1
                     },
                 }
             }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestCode = @"
+Public Class Class1
+#If DEBUG Then
+	Protected Overrides Sub Finalize()
+		System.Diagnostics.Debug.Fail(""Finalizer called!"")
+    End Sub
+#End If
+End Class
+",
+                SolutionTransforms =
+                {
+                    (solution, projectId) =>
+                    {
+                        var project = solution.GetProject(projectId);
+                        var parseOptions = ((VisualBasicParseOptions)project.ParseOptions).WithPreprocessorSymbols(new KeyValuePair<string, object>("DEBUG", true));
+
+                        return project.WithParseOptions(parseOptions)
+                            .Solution;
+                    },
+                }
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(1241, "https://github.com/dotnet/roslyn-analyzers/issues/1241")]
+        public async Task CA1821_DebugFailAndReleaseDirective_NoDiagnostic_FalsePositive()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+public class Class1
+{
+#if RELEASE
+	// There should be a diagnostic because in release the finalizer will be empty.
```

Comment:
Where is the issue tracking this failure?

Reply:
It's the issue #1241 mentioned in the `WorkItem`.

Reply:
Thanks, I didn't realize the issue was still open